todos/src/actions/index.js

```diff
@@ -52,37 +102,56 @@ export const editTodo = (id, onEdit) => {
   };
 };
 
-export const editUpdateCompleted = (
-  id,
-  isCompleted,
-  date,
-  completedDate,
-  onEdit
-) => {
-  return {
-    type: "EDIT_UPDATE_COMPLETED",
-    id,
-    isCompleted,
-    date,
-    completedDate,
-    onEdit,
+export const editUpdateCompleted =
+  (id, isCompleted, date, completedDate, onEdit) => async (dispatch) => {
+    try {
+      dispatch(setEditCardLoadingState(true));
+      const response = await updateStateTodoSupabaseOnEdit(
+        id,
+        isCompleted,
+        completedDate,
+        !onEdit
+      );
+      const tableData = response.data;
+      dispatch({
+        type: "EDIT_UPDATE_COMPLETED",
+        id: tableData.id,
+        isCompleted: tableData.isCompleted,
+        date: tableData.date,
+        completedDate: tableData.completedDate,
+        onEdit: tableData.onEdit,
+      });
+      successMessage(STATE_CHANGE_MESSAGE);
+    } catch (error) {
+      dangerMessage(WENT_WRONG_MESSAGE);
+    } finally {
+      dispatch(setEditCardLoadingState(false));
+    }
   };
-};
 
-export const updateTodo = (id, data, onEdit) => {
-  return {
-    type: "UPDATE_TODO",
-    id,
-    data,
-    onEdit,
-  };
+export const updateTodo = (id, data, onEdit, Toastflag) => async (dispatch) => {
+  try {
+    dispatch(setEditCardLoadingState(true));
+    const response = await updateTodoSupabase(id, data, !onEdit);
+    const tableData = response.data;
+    dispatch({
+      type: "UPDATE_TODO",
+      id: tableData.id,
+      data: tableData.data,
+      onEdit: tableData.onEdit,
+    });
+    Toastflag && successMessage(EDIT_SUCCESS_MESSAGE);
```

Comment:
What is ToastFlag? Is it a class or a component? If not we can make is cameCased rather than PascalCased.

Reply:
That's just a boolean value getting as props to decide whether to show the Edited Toast message or not. Not a component, so I'm renaming it as `toastFlag`.